pyproject.toml

```diff
@@ -1,160 +1,161 @@
-    # "N815",  # Variable in class scope should not be mixedCase  (uncomment if you want to allow mixedCase variable names in class scope)
-    # "N816",  # Variable in global scope should not be mixedCase  (uncomment if you want to allow mixedCase variable names in global scope)
-    ]
-line-length = 105
-select = [
-    "E",
-    "D",
-    "F",
-    "N",
-    "W",
-    "I",
-    "B",
-]
-target-version = "py39"
-
-[tool.ruff.pep8-naming]
-ignore-names = [
-    "test_*",
-    "setUp",
-    "tearDown",
-]
-
-[tool.ruff.pydocstyle]
-convention = "numpy"
-
-[tool.ruff.per-file-ignores]
-"__init__.py" = ["I001"]
-"./tests/*" = ["D"]
-
-[tool.pyright]
-exclude = [
-    ".git",
-    ".venv",
-    ".tox",
-    "build",
-    "dist",
-    "**/__pycache__",
-    "./docs/conf.py",
-    "./venv",
-]
-extraPaths = ["./src"]
-typeCheckingMode = "basic"
-useLibraryCodeForTypes = true
-# Activate the following rules step by step to (step by step..) improve code quality
-reportMissingParameterType = "error"
-reportUnknownParameterType = "warning"
-reportUnknownMemberType = "warning"
-reportMissingTypeArgument = "error"
-reportPropertyTypeMismatch = "error"
-reportFunctionMemberAccess = "warning"
-reportPrivateUsage = "warning"
-reportTypeCommentUsage = "warning"
-reportIncompatibleMethodOverride = "warning"
-reportIncompatibleVariableOverride = "error"
-reportInconsistentConstructor = "error"
-reportOverlappingOverload = "warning"
-reportUninitializedInstanceVariable = "warning"
-reportCallInDefaultInitializer = "warning"
-reportUnnecessaryIsInstance = "information"
-reportUnnecessaryCast = "warning"
-reportUnnecessaryComparison = "warning"
-reportUnnecessaryContains = "warning"
-reportUnusedCallResult = "warning"
-reportUnusedExpression = "warning"
-reportMatchNotExhaustive = "warning"
-reportShadowedImports = "warning"
-reportUntypedFunctionDecorator = "warning"
-reportUntypedBaseClass = "error"
-reportUntypedNamedTuple = "warning"
-
-# Activate the following rules only locally and temporary, i.e. for a QA session.
-# (For server side CI they are considered too strict.)
-# reportConstantRedefinition = "warning"
-# reportUnnecessaryTypeIgnoreComment = "information"
-# reportImportCycles = "warning"
-# reportImplicitStringConcatenation = "warning"
+[tool.poetry]
+name = "trafficgen"
+version = "0.2.0"
+description = "Automatic generation of ship traffic situations from a set of predefined situations"
+authors = ["Tom Arne Pedersen <Tom.Arne.Pedersen@dnv.com>"]
+license = "License stuff here" # FIXME
+readme = "README.md"
+
+[tool.poetry.dependencies]
+python = "^3.9"
+click = "^8.1"
+click-log = "^0.4.0"
+numpy = "^1.25.2"
+matplotlib = "^3.7.2"
+basemap = "^1.3.8"
+basemap-data-hires = "^1.3.2"
+global-land-mask = "^1.0.0"
+folium = "^0.14.0"
+pydantic = "^2.5"
+maritime-schema = { git = "https://github.com/dnv-opensource/maritime-schema.git" }

```

Comment:
change to `maritime-schema = "^0.0.3"` once the maritime-schema package is published.
(This line is the reason why the GitHub workflow fails. The workflow does not have permission to clone the maritime-schema repository)

Reply:
also done :)

Reply:
Installing the ship-traffic-generator package on my own machine I found out that we also need to limit the python versions we allow.

`python = "^3.9"`
needs to be changed to
`python = ">=3.9, <3.13"`

This is needed because "^3.9" effectively means ">=3.9, <4.0" , and hence includes 3.13
There are multiple other dependencies, though, that require python <3.13  (which makes sense. Also we ourselves would want to require that). 
If we implicitly allow python 3.13 (which "^3,9" does), then dependency resolution fails.